### PR TITLE
Getting Travis and Appveyor to run tests and trigger codecov

### PR DESCRIPTION
**Pull request by Aariq**
Because the `NOT_CRAN` environment variable is used by `testthat` to handle `skip_on_cran()`, Travis and Appveyor were not running tests because that environment variable was not set.  I also tweaked some things to get `rJava` (a dependency for `rcdk`) to install and load correctly on CI.

Having CI run tests required some restructuring of the tests themselves.

- added pinging of all services and skip tests if the service is down
- added `skip_on_travis()` and `skip_on_appveyor()` to all ChemSpider functions since CI systems don't have an API key (there may be a way to share a secure key with Travis, but we can figure that out later possibly)
- reorganized tests of utils to group by service so it's easier to skip when service is down

Other changes in tests:

- I changed the tests of `cs_control()` quite a bit.  Previously, they were using "Oxygen" which only had two results, so the order of the CSIDs could have easily been correct just by chance
- `cir_query()` was using a deprecated argument in tests
- Got rid of `context()` as it is no longer recommended by `testthat`
- Changed `expect_is()` to either `expect_type()` or `expect_s3_object()` as recommended by the help file for `expect_is()`
- changed `expect_equal(length(....` to `expect_length()`

Minor changes to code:

- Properly deprecated `first=` in `cir_query()`

Currently there are a few failing tests (some only on Windows), but I didn't want to change code too much in this PR.

PR task list:
- [ ] Update NEWS
- [ ] Add tests (if appropriate)
- [ ] Update documentation with `devtools::document()`
- [ ] Check package passed

### CODE REVIEW

**Comment by Aariq** [comment]
BTW, this won't actually trigger codecov until all the tests are passing, but when I skipped failing tests on my fork, it did actually produce a codecov report: https://codecov.io/gh/Aariq/webchem/branch/codecov

**Comment by stitam** [comment]
Also let's merge this PR before PR #243.

File: tests/testthat/test-extractors.R [hunk 1]
```diff
@@ -1,79 +1,116 @@
-context('extractors')
+require(RCurl)
+is.down <- function(qurl){
+  cont <- try(getURL(qurl, .encoding = 'UTF-8', .opts = list(timeout = 3)),
+              silent = TRUE)
+  return(inherits(cont, 'try-error'))
+}
```
[the remark this answers is not in the document]
Comment (stitam) [reply tests/testthat/test-extractors.R:6]: In fact, we already have issue #118 for this.

[the remark this answers is not in the document]
Comment (Aariq) [reply tests/testthat/test-extractors.R:6]: Good call.  I'll write these as `ping_` functions instead.

**Comment by Aariq** [comment]
Ok, failing tests are now skipped.  After merging this, might be a good idea to look into why they are failing and open issues.  I didn't really try to figure out why they were failing.

**Comment by stitam** [comment]
Let's reach a conclusion on exporting ping functions and after that I think this PR can be merged.

**Comment by stitam** [comment]
As this PR affects many other files, I added the "high priority" label to mark that it should be merged before the others.

**Comment by Aariq** [comment]
A few things:

1. chemid functions are not working correctly.  I just skipped all the tests for now.

2. Should the pubchem PUG-REST and PUG-VIEW ping functions be kept separate?  Right now `ping_service("pc")` checks if either `ping_pubchem` or `ping_pubchem_pw` returns FALSE.

3. I made `ping_service("cs_web")` only so we don't have to use the API so much in tests.  Right now the tests are eating up my API allowance very rapidly.  The alternative, I think, is to just include `ping_service("cs")` in every `test_that()` call.

**Comment by stitam** [comment]
> Should the pubchem PUG-REST and PUG-VIEW ping functions be kept separate? Right now ping_service("pc") checks if either ping_pubchem or ping_pubchem_pw returns FALSE.

I agree with your solution, it's best to keep it simple.

> I made ping_service("cs_web") only so we don't have to use the API so much in tests. Right now the tests are eating up my API allowance very rapidly. The alternative, I think, is to just include ping_service("cs") in every test_that() call.

I have the same issue:/

**Pull request merged**